tests/main/selinux-lxd/task.yaml

```diff
@@ -0,0 +1,59 @@
+summary: Check that LXD snap works on a system using SELinux in enforcing mode
+
+description: |
+    Make sure that LXD snap can be installed and used on a SELinux enable
+    system, with enforcing mode on.
+
+# Systems with SELinux enabled out of the box.
+systems: [fedora-*, centos-*]
+
+# lxd downloads can be quite slow
+kill-timeout: 25m
+
+# Start before anything else as it can take a really long time.
+priority: 1000
+
+prepare: |
+    getenforce > enforcing.mode
+    # Enable enforcing mode, our policy is already marked as permissive, so we
+    # will get audit entries but the program will not be stopped by SELinux
+    setenforce 1
+    ausearch --checkpoint stamp -m AVC || true
+
+restore: |
+    setenforce "$(cat enforcing.mode)"
+    rm -f stamp enforcing.mode
+
+execute: |
+    snap install lxd
+    ausearch --checkpoint stamp --start checkpoint -m AVC 2>&1 | MATCH 'no matches'
+
+    echo "Create a trivial container using the lxd snap"
+    snap set lxd waitready.timeout=240
+    lxd waitready
+    lxd init --auto
+
+    echo "Setting up proxy for lxc"
+    if [ -n "${http_proxy:-}" ]; then
+        lxd.lxc config set core.proxy_http "$http_proxy"
+    fi
+    if [ -n "${https_proxy:-}" ]; then
+        lxd.lxc config set core.proxy_https "$http_proxy"
+    fi
+
+    lxd.lxc launch "ubuntu:18.04" my-ubuntu
+
+    echo "Ensure we can run things inside"
+    lxd.lxc exec my-ubuntu echo hello | MATCH hello
+
+    echo "Stop the container"
+    lxd.lxc stop --force my-ubuntu
+
+    snap remove lxd
```

Comment:
Should we do this in restore, and also remove the container? (see the existing lxd test I mentioned earlier).

Reply:
Interstingly, the `tests/main/lxd` test does not remove the containers it creates. Perhaps what the regression test does is unnecessary?

Reply:
IMHO we should remove the containers not to have leftovers that could affect other tests. At the moment they are probably wiped from the disk on lxd snap removal, but with the automatic snapshots we will be needlessly creating a snapshot here.

Reply:
Pushed an update

Reply:
Thanks!